File: app/src/main/java/me/sargunvohra/android/purduediningcourts/base/BaseFragment.java
package me.sargunvohra.android.purduediningcourts.base;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.hannesdorfmann.fragmentargs.FragmentArgs;

import butterknife.ButterKnife;

public abstract class BaseFragment extends Fragment {
    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        FragmentArgs.inject(this);
    }

    public abstract int getLayoutRes();

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        super.onCreateView(inflater, container, savedInstanceState);
        View view = inflater.inflate(getLayoutRes(), container, false);
        ButterKnife.inject(this, view);
        return view;
    }
}

File: app/src/main/java/me/sargunvohra/android/diningcourts/base/BaseFragment.kt
package me.sargunvohra.android.diningcourts.base

import android.os.Bundle
import com.hannesdorfmann.fragmentargs.FragmentArgs
import com.karumi.rosie.view.RosieFragment
import org.jetbrains.anko.AnkoLogger

abstract class BaseFragment : RosieFragment(), AnkoLogger {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        FragmentArgs.inject(this)
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
    }
}